pineappl_cli/src/import.rs

```diff
@@ -0,0 +1,633 @@
+            ffi::GetBinSize(file_as_table)
+        } else {
+            vec![1.0; bins]
+        };
+        limits.reserve(2 * dimensions * bins);
+
+        for i in 0..bins {
+            for j in 0..dimensions {
+                let pair = ffi::GetObsBinDimBounds(
+                    file_as_table,
+                    i.try_into().unwrap(),
+                    j.try_into().unwrap(),
+                );
+
+                limits.push((pair.first, pair.second));
+            }
+        }
+
+        result.set_remapper(BinRemapper::new(normalizations.clone(), limits).unwrap())?;
+
+        Ok(result)
+    }
+}
+
+/// Converts fastNLO tables to PineAPPL grids.
+#[derive(Parser)]
+pub struct Opts {
+    /// Path to the input grid.
+    #[clap(parse(from_os_str), value_hint = ValueHint::FilePath)]
+    input: PathBuf,
+    /// Path to the converted grid.
+    #[clap(parse(from_os_str), value_hint = ValueHint::FilePath)]
+    output: PathBuf,
+    /// LHAPDF id or name of the PDF set to check the converted grid with.
+    #[clap(validator = helpers::validate_pdfset)]
+    pdfset: String,
+    /// LO coupling power in alpha.
+    #[clap(default_value_t = 0, long)]
+    alpha: u32,
+    /// Relative threshold between the table and the converted grid when comparison fails.
+    #[clap(default_value = "1e-10", long)]
+    accuracy: f64,
+    /// Prevents fastNLO from printing output.
+    #[clap(long = "silence-fastnlo")]
+    silence_fastnlo: bool,
+}
+
+impl Subcommand for Opts {
+    #[cfg(feature = "fastnlo")]
+    fn run(&self) -> Result<()> {
+        use lhapdf::Pdf;
+        use pineappl_fastnlo::ffi::{self, Verbosity};
+        use prettytable::{cell, row};
+
+        if self.silence_fastnlo {
+            ffi::SetGlobalVerbosity(Verbosity::SILENT);
+        }
+
+        let mut file = ffi::make_fastnlo_lhapdf_with_name_file_set(
+            self.input.to_str().unwrap(),
+            &self.pdfset,
+            0,
+            self.silence_fastnlo,
+        );
+        let grid = fastnlo::convert_fastnlo_table(&file, self.alpha)?;
+
+        let table_results = ffi::GetCrossSection(
+            ffi::downcast_lhapdf_to_reader_mut(file.as_mut().unwrap()),
+            false,
+        );
+        let pdf = self.pdfset.parse().map_or_else(
+            |_| Pdf::with_setname_and_member(&self.pdfset, 0),
+            Pdf::with_lhaid,
+        );
+        let results = helpers::convolute(&grid, &pdf, &[], &[], &[], 1);
+
+        let mut different = false;
+
+        let mut table = helpers::create_table();
+        table.set_titles(row![c => "b", "PineAPPL", "fastNLO", "rel. diff"]);
+
+        for (bin, (one, two)) in results
+            .into_iter()
+            .zip(table_results.into_iter())
+            .enumerate()
+        {
+            // catches the case where both results are zero
+            let rel_diff = if one == two {
+                0.0
+            } else {
+                (two / one - 1.0).abs()
```

Comment:
Are you sure we want `.abs()` here? The sign information might be relevant (of course it's not if it's `1e-14`, but it might be otherwise).

Reply:
This must be done, otherwise `rel_diff = -1.0` it would show up as agreement.

Reply:
Fixed in commit f236fdc7e9ab899f1aac1228b4426e875762dd53.